mining/worker.go

```diff
@@ -166,65 +160,83 @@ func (w *DefaultWorker) Mine(ctx context.Context, base types.TipSet, nullBlkCoun
 		return false
 	}
 
-	st, err := w.getStateTree(ctx, base)
-	if err != nil {
-		log.Errorf("Worker.Mine couldn't get state tree for tipset: %s", err.Error())
-		outCh <- Output{Err: err}
-		return false
-	}
-
 	log.Debugf("Mining on tipset: %s, with %d null blocks.", base.String(), nullBlkCount)
 	if ctx.Err() != nil {
 		log.Warningf("Worker.Mine returning with ctx error %s", ctx.Err().Error())
 		return false
 	}
 
-	challenge, err := consensus.CreateChallengeSeed(base, uint64(nullBlkCount))
+	// Read uncached worker address
+	workerAddr, err := w.api.MinerGetWorkerAddress(ctx, w.minerAddr)
 	if err != nil {
-		log.Warningf("Worker.Mine couldn't create challenge seed: %s", err.Error())
 		outCh <- Output{Err: err}
 		return false
 	}
-	prCh := createProof(challenge, w.createPoSTFunc)
 
-	// Read uncached worker address
-	workerAddr, err := w.api.MinerGetWorkerAddress(ctx, w.minerAddr)
+	// Create the next ticket.
+	// Derive from mining base's min ticket
+	prevTicket, err := base.MinTicket()
+	if err != nil {
+		log.Warningf("Worker.Mine couldn't read parent ticket")
+		outCh <- Output{Err: err}
+		return false
+	}
+	nextTicket, err := w.ticketGen.NextTicket(prevTicket, workerAddr, w.workerSigner, uint64(nullBlkCount))
 	if err != nil {
+		log.Warningf("Worker.Mine couldn't generate next ticket")
 		outCh <- Output{Err: err}
 		return false
 	}
 
-	var proof types.PoStProof
-	var ticket types.Ticket
+	// Provably delay for the blocktime
+	done := make(chan struct{})
+	errCh := make(chan error)
+	go func() {
+		defer close(done)
+		defer close(errCh)
+		err = w.ticketGen.NotarizeTime(&nextTicket)
+		if err != nil {
+			errCh <- err
+		}
+		// TODO #2223 remove this explicit wait if/when NotarizeTime calls VDF
+		w.clock.Sleep(w.api.BlockTime())
+		done <- struct{}{}
+	}()
+
 	select {
 	case <-ctx.Done():
 		log.Infof("Mining run on base %s with %d null blocks canceled.", base.String(), nullBlkCount)
 		return false
-	case prChRead, more := <-prCh:
-		if !more {
-			log.Errorf("Worker.Mine got zero value from channel prChRead")
-			return false
-		}
-		proof := append(types.PoStProof{}, prChRead[:]...)
-		ticket, err = consensus.CreateTicket(proof, workerAddr, w.workerSigner)
-		if err != nil {
-			log.Errorf("failed to create ticket: %s", err)
-			return false
-		}
+	case <-done: //fallthrough
```

Comment:
This is confusing, I don't think we want to fallthrough here. I believe we intend to break, which is implicit. If that's the case remove the comment.